fastparse/shared/src/test/scala/fastparse/iterator/IteratorTests.scala

```diff
@@ -0,0 +1,62 @@
+package fastparse.iterator
+
+import fastparse.all._
+import utest._
+
+object IteratorTests extends TestSuite {
```

Comment:
I don't think the unit tests we have here are sufficient; we should add some unit tests that (or modify these to) subclass `ParserInput` and hook into `dropBuffer` in order to verify that `dropBuffer` indeed gets called at exactly the right moments within these tests, and that the buffer does not grow past whatever size we think it should not grow past. 

That will let us be confident that we're actually being as streaming as we should be streaming, and not accidentally loading the whole thing into memory before dealing with it